ibis/backends/__init__.py

```diff
@@ -47,6 +47,31 @@ class TablesAccessor(collections.abc.Mapping):
     def __init__(self, backend: BaseBackend):
         self._backend = backend
 
+        # list_* methods that might exist on a given backends
+        list_methods = [
+            "list_tables",
+            "list_temp_tables",
+            "list_views",
+            "list_temp_views",
+        ]
+
+        self._tables = []
+        for method_name in list_methods:
+            self._tables.extend(self._execute_if_exists(method_name))
+
+        # TODO: maybe there is a better way of removing duplicates, for now
+        self._tables = list(set(self._tables))
```

Comment:
I think there are two things to consider here, one that is actionable now, one that is a TODO for later on:

1. (now) If `_tables` is only populated at backend initialization, then `con.tables` won't show any tables/views that a user creates using, e.g. `read_parquet` -- I think this needs to be computed on demand.
2. (later) If we show the _type_ of a table object in the output of `con.tables`, we might not want to deduplicate the results, but rather show that there is say a table called `foo` AND a view called `foo`

Reply:
>  1. (now) If _tables is only populated at backend initialization, then con.tables won't show any tables/views that a user creates using, e.g. read_parquet -- I think this needs to be computed on demand.

But isn't the computation being taken care by the fact that we have this method in the Backend class? 

```python
    @functools.cached_property
    def tables(self):
    ...
```

As a matter of fact I just tried this with a file I have locally, and it shows up: 

```python
In [1]: import ibis

In [2]: con = ibis.get_backend()

In [3]: t = con.read_parquet("docs/posts/ibis-lonboard-geo/overture_maps_data/download_dc/dc-infra.geoparquet", "my_geo_table")

In [4]: con.tables
Out[4]: 
Tables
------
- my_geo_table
```

For 2) I wonder if we want to do this now, or rethink the repr later on a followup

Reply:
The 2nd point can definitely happen in a follow-up, I just think we should leave a `TODO` there.

Re: the first point, no, I think this is not correct, because tables that are created _after_ the initial instantiation (triggered by asking for `con.tables`) won't be picked up, because the `self._tables` attribute in the `Accessor` won't get re-initialized.

e.g.:

```python
[ins] In [1]: import ibis

[ins] In [2]: con = ibis.duckdb.connect()

[ins] In [3]: con.create_table("table1", {"col": [["abc"], ["abcd"]]})
Out[3]: 
DatabaseTable: memory.main.table1
  col array<string>

[ins] In [4]: con.tables
Out[4]: 
Tables
------
- table1

[nav] In [5]: con.create_table("table2", {"col": [["abc"], ["abcd"]]})
Out[5]: 
DatabaseTable: memory.main.table2
  col array<string>

[ins] In [6]: con.tables
Out[6]: 
Tables
------
- table1
```

Reply:
>  The 2nd point can definitely happen in a follow-up, I just think we should leave a TODO there.

This makes sense, I'll do that. 

> Re: the first point, no, I think this is not correct, because tables that are created after the initial instantiation (triggered by asking for con.tables) won't be picked up, because the self._tables attribute in the Accessor won't get re-initialized.

Oh, I see thank for the clarification.  One solution is  make the tables method  a `@property` instead of `@functools.cached_property`. Is that what you meant by "I think this needs to be computed on demand"

Reply:
That will suffer from the same issue.  The `self._tables` attribute is being populated in the `__init__` of the class -- once the Accessor is initialized, then `_tables` is frozen and doesn't get updated.

Reply:
Oh, maybe I misunderstood what you were saying -- if `_tables` is a `property` in the `TablesAccessor` class then that will work

Reply:
I was referring to the `tables` method, and by doing that your example now shows both tables. That being said idk if that's by happy accident. 

#this code no is adapted to reflect Phillips PR
```python
    @property #I meant this
    def tables(self):
        """An accessor for tables in the database.
        ...
        """
        return TablesAccessor(weakref.proxy(self))
```

```python
In [1]: import ibis

In [2]: con = ibis.duckdb.connect()

In [3]: con.create_table("table1", {"col": [["abc"], ["abcd"]]})
Out[3]: 
DatabaseTable: memory.main.table1
  col array<string>

In [4]: con.tables
Out[4]: 
Tables
------
- table1

In [5]: con.create_table("table2", {"col": [["abc"], ["abcd"]]})
Out[5]: 
DatabaseTable: memory.main.table2
  col array<string>

In [6]: con.tables
Out[6]: 
Tables
------
- table1
- table2
```

Reply:
Ok, that works but it's creating a new `TablesAccessor` object each time.  I don't know if we care about that overhead, but I was thinking that the thing that should be a `property` was `_tables`, and then the above `tables` can remain a `cached_property`